src/categorical_algebra/DataMigrations.jl

```diff
@@ -161,6 +162,30 @@ function migrate!(X::ACSet, Y::ACSet, FOb, FHom)
   migrate!(X, Y, M)
 end
 
+function (::Type{T})(X::ACSet, F::FinFunctor) where T <: ACSet
+  Base.depwarn("Data migration constructor is deprecated, use `migrate` instead", :ACSet)
+  migrate(T, X, F)
+end
+
+function (::Type{T})(X::ACSet, FOb::AbstractDict, FHom::AbstractDict) where T <: ACSet
+  Base.depwarn("Data migration constructor is deprecated, use `migrate` instead", :ACSet)
+  migrate(T, X, FOb, FHom)
+end
+
+# On morphisms
+
+function migrate(::Type{T}, f::TightACSetTransformation, FOb::AbstractDict, FHom::AbstractDict) where T <: ACSet
+  d = Dict()
+  for (ob_dom,ob_codom) in pairs(FOb)
+    if Symbol(ob_codom) ∈ keys(components(f))
+      d[Symbol(ob_dom)] = f[Symbol(ob_codom)]
+    end
+  end
+  Fd, Fcd = migrate(T, dom(f), FOb, FHom), migrate(T, codom(f), FOb, FHom)
+  TightACSetTransformation(NamedTuple(d), Fd, Fcd)
+end
+
+
```

Comment:
Shouldn't there be methods that let you migrate using a `Functor` or a `DataMigration` too?

Reply:
More generally it would probably be nice to have the data flow for migrating morphisms match that for migrating ACSets, so eg having the `FOb,FHom` API go through the `Migration` API rather than short-circuiting straight to the end. If you wrote it this way with performance in mind, I'd say that users ought to build the migration once and for all if they're going to be migrating a lot along the same migration.

Reply:
Other than that, looks good; my only other thought is whether it might be nice to have a method `migrate(f,M,X,Y)` where `X` and `Y` are *already migrated* to the target category. This would be noticeably faster if you were migrating many morphisms among few acsets at some point. And the more usual `migrate` method could just migrate `X` and `Y` and then call into this one.

Reply:
I'll add the things that didn't get covered in this PR as an issue to be addressed after MigrationFest2023!